test/external/ringtest/CMakeLists.txt

```diff
@@ -126,7 +126,53 @@ foreach(processor cpu gpu)
     ENVIRONMENT OMP_NUM_THREADS=${ringtest_num_threads} NEURON_INIT_MPI=1
     COMMAND ${ringtest_python} -tstop 100 -coreneuron -nt ${ringtest_num_datasets} ${gpu_arg})
 endforeach()
+
+nrn_add_test(
+  GROUP external_ringtest
+  NAME coreneuron_mpi_registermapping_directmode
+  REQUIRES coreneuron mpi python
+  PROCESSORS ${ringtest_mpi_ranks}
+  ENVIRONMENT NEURON_INIT_MPI=1 LIBSONATA_ZERO_BASED_GIDS=1
+  COMMAND ${ringtest_special} -tstop 100 -coreneuron -registermapping)
+nrn_add_test(
+  GROUP external_ringtest
+  NAME coreneuron_mpi_registermapping_filemode
+  REQUIRES coreneuron mpi python
+  PROCESSORS ${ringtest_mpi_ranks}
+  ENVIRONMENT NEURON_INIT_MPI=1 LIBSONATA_ZERO_BASED_GIDS=1
+  COMMAND ${ringtest_special} -tstop 100 -coreneuron -registermapping -filemode)
+
 # Step 3 -- add a job that compares the outputs of all the tests added in Step 2
 nrn_add_test_group_comparison(
   GROUP external_ringtest
   REFERENCE_OUTPUT asciispikes::external/tests/ringtest/reference_data/spk1.100ms.std.ref)
+
+if(CORENRN_ENABLE_REPORTING)
+  find_program(H5DIFF_EXECUTABLE "h5diff")
+  set(reference_path
+      "${PROJECT_BINARY_DIR}/test/external/tests/ringtest/reference_data/soma.ref.h5")
+  cpp_cc_build_time_copy(
+    INPUT "${PROJECT_SOURCE_DIR}/external/tests/ringtest/reference_data/soma.ref.h5"
+    OUTPUT "${reference_path}")
+  foreach(testfolder coreneuron_mpi_registermapping_directmode
+                     coreneuron_mpi_registermapping_filemode)
+    if(H5DIFF_EXECUTABLE)
+      add_test(NAME external_ringtest::compare_soma_report_${testfolder}
+               COMMAND "h5diff" ${PROJECT_BINARY_DIR}/test/external_ringtest/${testfolder}/soma.h5
+                       ${reference_path})
+    else()
+      add_test(
+        NAME external_ringtest::compare_soma_report_${testfolder}
+        COMMAND
+          python -c
+          "import h5py; \
+          file = h5py.File(\"${PROJECT_BINARY_DIR}/test/external_ringtest/${testfolder}/soma.h5\"); \
+          ref_file = h5py.File(\"${reference_path}\"); \
+          data = set(file[\"/report/default/mapping/node_ids\"]); \
+          ref_data = set(ref_file[\"/report/default/mapping/node_ids\"]); \
+          assert data == ref_data;")
```

Comment:
@WeinaJi : here in the `else` block we have an assumption that `h5py` is available. This may not be the case on the developers' machine.

I wonder what's better: not have `else()` block. If `h5diff` is missing then just throw a warning and skip the test?